app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CalendarViewTest.java

```diff
@@ -0,0 +1,139 @@
+        return activity[0];
+    }
+
+    @Rule
+    public ActivityScenarioRule<MainScreenActivity> mainScreenActivityActivityScenarioRule =
+            new ActivityScenarioRule<>(MainScreenActivity.class);
+
+    @Test
+    public void attachmentCorrectlyShows() {
+        onView(withId(R.id.house_imageButton)).perform(click());
+        onView(withId(R.id.housesView)).perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        onView(withId(R.id.calendar))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(0, RecyclerViewHelperActions.clickChildViewWithId(R.id.event_upcoming_attach)));
+        onView(withText("Show")).perform(click());
+        onView(withId(R.id.image_dialog)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+    }
+
+    @Test
+    public void removeAttachmentCorrectlyWorks() {
+        onView(withId(R.id.house_imageButton)).perform(click());
+        onView(withId(R.id.housesView)).perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        onView(withId(R.id.calendar))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(2, RecyclerViewHelperActions.clickChildViewWithId(R.id.event_upcoming_attach)));
+        onView(withText("Remove")).perform(click());
+        storage.getReference().child("to_delete_attachment.jpg").getDownloadUrl().addOnCompleteListener(
+                task -> assertThat(task.isSuccessful(), is(false))
+        );
+    }
+
+    @Test
+    public void attachCorrectlyFiresIntent() {
+        Intents.init();
+        onView(withId(R.id.house_imageButton)).perform(click());
+        onView(withId(R.id.housesView)).perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        onView(withId(R.id.calendar))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(1, RecyclerViewHelperActions.clickChildViewWithId(R.id.event_upcoming_attach)));
+        onView(withText("Attach")).perform(click());
+        intended(hasAction(Intent.ACTION_GET_CONTENT));
+        Intents.release();
+    }
+
+    @Test
+    public void showCorrectlyToastsOnNoAttachment() {
+        onView(withId(R.id.house_imageButton)).perform(click());
+        onView(withId(R.id.housesView)).perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        onView(withId(R.id.calendar))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(1, RecyclerViewHelperActions.clickChildViewWithId(R.id.event_upcoming_attach)));
+        onView(withText("Show")).perform(click());
+
+        try {
+            onView(withText("Could not find the attachment")).
+                    inRoot(withDecorView(
+                            not(is(getCurrentActivity()
+                                    .getWindow().getDecorView())))).
+                    check(matches(isDisplayed()));
+        } catch (Throwable throwable) {
+            throwable.printStackTrace();
+        }
```

Comment:
I think that this checks if the text is displayed but that is black magic xD
Maybe add information with a comment so we would b eable to do it again if needed ?

Reply:
Adding comments to the code, this checks that a toast is displayed somehow